src/python_testing/matter_testing_support.py

```diff
@@ -963,43 +963,51 @@ def get_test_desc(self, test: str) -> str:
         except AttributeError:
             return test
 
-    # Sends an out-of-band command to a Matter app
-    def write_to_app_pipe(self, command):
+    def write_to_app_pipe(self, app_pipe_name: str, command_dict: dict):
         """
+        Sends an out-of-band command to a Matter app.
+
         Use the following environment variables:
 
          - LINUX_DUT_IP 
-            * if not provided, the Matter app is assumed to run on the same machine as this test,
+            * if not provided, the Matter app is assumed to run on the same machine as the test,
               such as during CI, and the commands are sent to it using a local named pipe
             * if provided, the commands for writing to the named pipe are forwarded to the DUT
-        - LINUX_DUT_UNAME
+        - LINUX_DUT_USER
             * if LINUX_DUT_IP is provided, use this for the DUT user name
-            * if LINUX_DUT_NAME is not provided, a default value is used ("root")
             * If a remote password is needed, set up ssh keys to ensure that this script can log in to the DUT without a password:
                  + Step 1: If you do not have a key, create one using ssh-keygen
                  + Step 2: Authorize this key on the remote host: run ssh-copy-id user@ip once, using your password
                  + Step 3: From now on ssh user@ip will no longer ask for your password
         """
+
+        if not isinstance(app_pipe_name, str):
```

Comment:
Given the `:str` and `:dict` in line 966, are these `if`s required?

Reply:
Those are type hints (rather than "constraints"), so yes.